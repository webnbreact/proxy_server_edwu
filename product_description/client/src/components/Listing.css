.button {
  color: #199094;
  font-weight: bold;
}

.button:hover {
  cursor: pointer;
  text-decoration: underline;
}

.containerDiv {
  color:#585a3a;
  line-height: 22px;
  font-family: 'Montserrat', sans-serif;
  margin: 0 15%;
  max-width: 510px;
}

.host_image {
  text-align: center;
  height: 64px;
  width: 64px;
  margin-left: 40px;
  display: inline-block;
  background-color: white;
  border-radius: 50%;
  border: 1x solid grey;
  padding:0px;
  overflow: hidden;
}

.host_image img {
  height: 100%;
  max-width: 100%;
  object-fit: cover;
}

.home_location {
  margin: 0px;
  word-wrap: break-word;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.375em;
  color: #484848;
}

.home_name {
  font-size: 32px;
  font-weight: 800;
  line-height: 1.125em;
  color: #484848;
  padding-top: 6px;
  padding-bottom: 6px;
}

.home_type {
  box-sizing: border-box;
  font-weight: 800;
  font-size: 12px;
  font-weight: bold;
  line-height: 1.3333333333333333em;
  text-transform: uppercase;
} 
