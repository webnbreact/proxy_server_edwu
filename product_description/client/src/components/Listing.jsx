import React from 'react';
import styles from './Listing.css';

import Amenities from './Amenities.jsx';

class Listing extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: 1,
      listingInfo: {},
      isExpanded: false
    }
    this.showMore = this.showMore.bind(this)
    this.renderMoreDescription = this.renderMoreDescription.bind(this)
  }

  componentDidMount() {
    fetch(`http://localhost:3003/rooms/${this.state.id}/listings`)
      .then(response =>
        response.json()
      )
      .then((result) => {
        this.setState({listingInfo: result})
      })
      .then(myJson => {
        console.log(`this is myJSON`, myJson, this.state.listingInfo)
      })
      .catch(error => {
        console.log('fetch error', error)
      })
  }

  showMore() {
    this.setState({
      isExpanded: !this.state.isExpanded,
    })
  }

  renderMoreDescription() {
    if (this.state.isExpanded) {
      return this.state.listingInfo.homeDescriptionMore
    } else {
      return;
    }
  }

  render() {
    return (    
      <div className={styles.containerDiv}>
        <div className={styles.host_image}>
          <img src={this.state.listingInfo.hostImage} />
        </div>
        <ul className={styles.home_type}>{this.state.listingInfo.homeType}</ul>
        <ul className={styles.home_name}>{this.state.listingInfo.homeName}</ul>
        <ul className={styles.home_location}>{this.state.listingInfo.homeLocation}</ul>
        <hr />
        <ul>{this.state.listingInfo.homeDescription}</ul>
        <ul>{this.renderMoreDescription()}</ul>
        <ul className={styles.button} onClick={this.showMore}>
        {this.state.isExpanded ? 'Hide' : 'Read more about the space'}</ul> 
        <ul className={styles.button}>Contact host</ul>
        <hr />
        <Amenities />
        <hr />
      </div>
    )
  }
}

export default Listing;
