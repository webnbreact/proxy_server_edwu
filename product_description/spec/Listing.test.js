import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Listing from '../client/src/components/Listing.jsx';

function setup() {
  const wrapper = shallow(<Listing />);
  return {wrapper};
}

describe('Listing Test', () => {
  it('shallow renders without crashing', () => {
    const component = shallow(<Listing />, { disableLifecycleMethods: true })
    expect(toJson(component)).toMatchSnapshot();
  });
});