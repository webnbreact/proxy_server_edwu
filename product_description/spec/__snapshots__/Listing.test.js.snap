// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing Test shallow renders without crashing 1`] = `
<div>
  <div>
    <div>
      <ul />
      <ul />
      <ul />
      <ul />
      <button
        onClick={[Function]}
      >
        Read more about the space
      </button>
    </div>
  </div>
</div>
`;

exports[`Listing Test should render 1`] = `<div />`;
